src/Field/CoordinateReferenceSystemCombo/CoordinateReferenceSystemCombo.jsx

```diff
@@ -0,0 +1,239 @@
+      value: PropTypes.string,
+      code: PropTypes.string
+    })),
+    /**
+     * A function
+     * @type {String}
+     */
+    onSelect: PropTypes.func
+  }
+
+  static defaultProps = {
+    emptyTextPlaceholderText: 'Please select a CRS',
+    crsApiUrl: 'https://epsg.io/',
+    onSelect: () => {}
+  }
+
+  /**
+   * Create a CRS combo.
+   * @constructs CoordinateReferenceSystemCombo
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      crsDefinitions: [],
+      value: null
+    };
+
+    this.onCrsItemSelect = this.onCrsItemSelect.bind(this);
+  }
+
+  /**
+   * Fetch CRS definitions from epsg.io for given search string
+   *
+   * @param {String} searchVal The search string
+   */
+  fetchCrs = (searchVal) => {
+    const { crsApiUrl } = this.props;
+
+    const queryParameters = {
+      format: 'json',
+      q: searchVal
+    };
+
+    return fetch(`${crsApiUrl}?${UrlUtil.objectToRequestString(queryParameters)}`)
+      .then(response => response.json());
+  }
+
+  /**
+   * This function gets called on success of the EPSG.io fetch.
+   * It simply calls callback function if defined
+   *
+   * @param {Function} callback The callback function to be called
+   * @param {Object} crsDefinitions Array of CRS definitons used in
+   *                 CoordinateReferenceSystemCombo
+   */
+  runViaCallback = (callback, crsDefinitions) => {
+    if (!callback) {
+      return false;
+    }
+    callback(crsDefinitions);
+  }
+
+  /**
+   * This function gets called when the EPSG.io fetch returns an error.
+   * It logs the error to the console.
+   *
+   * @param {String} error The error string.
+   */
+  onFetchError(error) {
+    Logger.error(`Error while requesting in CoordinateReferenceSystemCombo: ${error}`);
+  }
+
+  /**
+   * This function transforms results of EPSG.io
+   *
+   * @param {Object} json The result object of EPSG.io-API, see where
+   *                 https://github.com/klokantech/epsg.io#api-for-results   *
+   * @return {Array} Array of CRS definitons used in CoordinateReferenceSystemCombo
+   */
+  transformResults = (json) => {
+    const results = json.results;
+    if (results && results.length > 0) {
+      return results.map(obj => ({code: obj.code, value: obj.name, proj4def: obj.proj4, bbox: obj.bbox}));
+    } else {
+      return [];
+    }
+  }
+
+  /**
+   * This function gets called when the EPSG.io fetch returns an error.
+   * It logs the error to the console.
+   *
+   * @param {String} error The error string.
+   */
+  handleSearch = (value) => {
+    const {
+      predefinedCrsDefinitions
+    } = this.props;
+
+    if (!value || value.length === 0) {
+      this.setState({
+        value,
+        crsDefinitions: []
+      });
+      return;
+    }
+
+    if (!predefinedCrsDefinitions) {
+      this.fetchCrs(value)
+        .then(this.transformResults)
+        .then(this.runViaCallback.bind(this, crsDefinitions => this.setState({ crsDefinitions, value })))
```

Comment:
I don't get the sense in the `runViaCallback` method.
```javascript
.then(this.runViaCallback.bind(this, crsDefinitions => this.setState({ crsDefinitions, value })))
```
…should be…

```javascript
.then(crsDefinitions => this.setState({crsDefinitions})) //  value is already set in line 149
```

Reply:
removed in 17775ed